localstack/services/events/provider_v2.py

```diff
@@ -58,18 +148,185 @@ def put_rule(
         event_bus_name: EventBusNameOrArn = None,
         **kwargs,
     ) -> PutRuleResponse:
-        rule = {
-            "Name": name,
-            "ScheduleExpression": schedule_expression,
-            "EventPattern": event_pattern,
-            "State": state,
-            "Description": description,
-            "RoleArn": role_arn,
-            "EventBusName": event_bus_name,
-        }
-        self._rules[name] = rule
-
-        response = PutRuleResponse(
-            RuleArn=f"arn:aws:events:{context.region}:{context.account_id}:rule/{name}",
+        region = context.region
+        account_id = context.account_id
+        event_bus_name = self._extract_event_bus_name(event_bus_name)
+        store = self.get_store(context)
+        event_bus = self.get_event_bus(event_bus_name, store)
+        existing_rule = event_bus.rules.get(name)
+        targets = existing_rule.targets if existing_rule else None
+        # TODO use _get_rule_worker and add logic to auto create rule worker if not exist
+        rule_worker = RuleWorker(
+            name,
+            region,
+            account_id,
+            schedule_expression,
+            event_pattern,
+            state,
+            description,
+            role_arn,
+            tags,
+            event_bus_name,
+            targets,
         )
+        self._rule_workers[rule_worker.arn] = rule_worker
+        event_bus.rules[name] = rule_worker.rule
+        response = PutRuleResponse(RuleArn=rule_worker.arn)
         return response
+
+    @handler("DeleteRule")
+    def delete_rule(
+        self,
+        context: RequestContext,
+        name: RuleName,
+        event_bus_name: EventBusNameOrArn = None,
+        force: Boolean = None,
+        **kwargs,
+    ) -> None:
+        store = self.get_store(context)
+        event_bus_name = self._extract_event_bus_name(event_bus_name)
+        event_bus = self.get_event_bus(event_bus_name, store)
+        if rule := event_bus.rules.get(name):
+            rule_worker = self._rule_workers[rule.arn]
+            rule_worker.delete()
+            self._rule_workers.pop(name)
```

Comment:
I guess one of the tricky parts is going to be how to propagate such updates into potentially multiple thread workers in a safe way.